### PR TITLE
fix(artifact_proxy): add missing Windows artifact patterns to config.dart

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds missing artifact patterns for Windows to artifact_proxy's config.dart.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/artifact_proxy/lib/config.dart [hunk 1]
```diff
@@ -22,6 +26,7 @@ final engineArtifactPatterns = {
   r'flutter_infra_release\/flutter\/(.*)\/ios-release\/artifacts\.zip',
   r'flutter_infra_release\/flutter\/(.*)\/ios-profile\/artifacts\.zip',
   r'flutter_infra_release\/flutter\/(.*)\/flutter-web-sdk-darwin-x64\.zip',
+  r'flutter_infra_release\/flutter\/(.*)\/flutter-web-sdk-windows-x64\.zip',
```
Comment (eseidel) [inline packages/artifact_proxy/lib/config.dart:29]: We're presumably missing the -linux variant of this too. ☹️

File: packages/artifact_proxy/lib/config.dart [hunk 2]
```diff
@@ -84,6 +89,7 @@ final engineArtifactPatterns = {
   r'download.flutter.io\/io\/flutter\/arm64_v8a_release\/1.0.0-(.*)\/arm64_v8a_release-1.0.0-(.*)\.pom',
   r'download.flutter.io\/io\/flutter\/arm64_v8a_release\/1.0.0-(.*)\/arm64_v8a_release-1.0.0-(.*)\.jar\.sha1',
   r'download.flutter.io\/io\/flutter\/arm64_v8a_release\/1.0.0-(.*)\/arm64_v8a_release-1.0.0-(.*)\.jar',
+  r'download.flutter.io\/io\/flutter\/x86_64_release/\1.0.0-(.*)\/x86_64_release-1.0.0-(.*)\.pom',
```
Comment (eseidel) [inline packages/artifact_proxy/lib/config.dart:92]: You'll end up with the other 3 patterns to as soon as you do an Android build I think?

**Review by eseidel — COMMENTED** [review]
I'm surprised by the .pom without .jar which I think you'll want to fix?  But otherwise lgtm.

**Review by eseidel — APPROVED** [review]

File: packages/artifact_proxy/lib/config.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/artifact_proxy/lib/config.dart:92]: Seemingly not? I'm able to run, create a release, and create a patch with this as it is 🤷

File: packages/artifact_proxy/lib/config.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/artifact_proxy/lib/config.dart:29]: :(

**Comment by eseidel** [comment]
the x86_64 versions are likely only used by the x86_64 android emulator.

**Pull request merged**